aten/src/ATen/native/quantized/cpu/fbgemm_utils.cpp

```diff
@@ -385,10 +387,72 @@ torch::class_<LinearPackedParamsBase> register_linear_params() {
   return register_linear_params;
 }
 
+enum class EmbeddingPackedParamType : int64_t {QEMBEDDING_BAG = 0};
+
+torch::class_<EmbeddingPackedParamsBase> register_embedding_params() {
+  // Type for __getstate__/__setstate__ serialization
+  //
+  // Element 0 is a enum of EmbeddingPackedParamType.
+  // Element 1 is the version of the PackedParam structure
+  // Element 2 is the Tensors contained in the Param instance
+  // Element 3 is the double values (if any) contained in the Param instance
+  // Element 4 is the int values (if any) contained in the Param instance
+
+  using EmbeddingParamsSerializationType = std::tuple<
+    int64_t,
+    int64_t,
+    std::vector<at::Tensor>,
+    std::vector<double>,
+    std::vector<int64_t>>;
+
+  static auto register_embedding_params =
+    torch::class_<EmbeddingPackedParamsBase>(
+      "quantized", "EmbeddingPackedParamsBase")
+      .def_pickle(
+          [](const c10::intrusive_ptr<EmbeddingPackedParamsBase>& params)
+              -> EmbeddingParamsSerializationType { // __getstate__ call
+            at::Tensor weight = params->unpack();
+            std::vector<at::Tensor> tensors_to_serialize = {weight};
+            std::vector<double> doubles_to_serialize = {};
+            int64_t bit_rate = params->bit_rate();
+            int64_t version = params->version();
+            std::vector<int64_t> longs_to_serialize = {bit_rate};
+            return EmbeddingParamsSerializationType(
+              static_cast<int64_t>(EmbeddingPackedParamType::QEMBEDDING_BAG),
```

Comment:
actually, what are the different types? maybe we can also encode this in the type of the type of the embeddingpackedparams itself? e.g. we have Conv2dPackedParams, Conv1dPackedParams. not suggesting we should do that, but I think we should consider both options.

Reply:
The types will be dependent on the tensor dtype, currently we have 8-bit and 4-bit (planned). From the op side I was planning to keep it same (eg. `quantized::embedding_bag_prepack`), since we can look at tensor dtype and decide if we should do 8-bit or 4-bit packing. 
Since we have one op, maybe having same name for the PackedParams might make more sense. Let me know if you feel that we should encode it in the name and also have different ops for 8-bit vs 4-bit (does it make a difference for graph mode?). 
The actual quantized op will be different i.e. quantized::embedding_bag_byte vs quantized::embedding_bag_4bit

Reply:
Are you talking about `bit_rate`? I'm talking about the EmbeddingPackedParamType here actually. Do you have a doc describing all the variations we plan to support?